ckan/lib/dictization/model_dictize.py

```diff
@@ -114,8 +114,8 @@ def resource_dictize(res, context):
                                     resource_id=res.id,
                                     filename=cleaned_name,
                                     qualified=True)
-    elif resource['url'] and not urlparse.urlsplit(url).scheme and not context.get('for_edit'):
-        resource['url'] = u'http://' + url.lstrip('/')
+    elif resource['url'] and not urlparse.urlsplit(url).scheme:
+        res_dict['url'] = res_dict['url'].rsplit('/')[-1]
```

Comment:
this returns just the file name part of any resource URL that is missing an scheme when either `for_edit` is true or the `url_type` is *not* upload (see if statement above). That doesn't make sense to me.

Reply:
now ive got lost a bit, the change that was requested a while ago, isnt been targeting this condition ?!